wrapper/src/main/java/software/amazon/jdbc/PartialPluginService.java

```diff
@@ -0,0 +1,768 @@
+  }
+
+  @Override
+  public HostSpec getCurrentHostSpec() {
+    if (this.currentHostSpec == null) {
+      this.currentHostSpec = this.initialConnectionHostSpec;
+
+      if (this.currentHostSpec == null) {
+        if (this.getAllHosts().isEmpty()) {
+          throw new RuntimeException(Messages.get("PluginServiceImpl.hostListEmpty"));
+        }
+
+        this.currentHostSpec = this.getWriter(this.getAllHosts());
+        final List<HostSpec> allowedHosts = this.getHosts();
+        if (!Utils.containsUrl(allowedHosts, this.currentHostSpec.getUrl())) {
+          throw new RuntimeException(
+              Messages.get("PluginServiceImpl.currentHostNotAllowed",
+                  new Object[] {
+                      currentHostSpec == null ? "<null>" : currentHostSpec.getUrl(),
+                      Utils.logTopology(allowedHosts, "")})
+          );
+        }
+
+        if (this.currentHostSpec == null) {
+          this.currentHostSpec = this.getHosts().get(0);
+        }
+      }
+      if (this.currentHostSpec == null) {
+        throw new RuntimeException("Current host is undefined.");
+      }
+      LOGGER.finest(() -> "Set current host to " + this.currentHostSpec);
+    }
+    return this.currentHostSpec;
+  }
+
+  public void setInitialConnectionHostSpec(final @NonNull HostSpec initialConnectionHostSpec) {
+    this.initialConnectionHostSpec = initialConnectionHostSpec;
+  }
+
+  @Override
+  public HostSpec getInitialConnectionHostSpec() {
+    return this.initialConnectionHostSpec;
+  }
+
+  @Override
+  public String getOriginalUrl() {
+    return this.originalUrl;
+  }
+
+  @Override
+  @Deprecated
+  public void setAllowedAndBlockedHosts(AllowedAndBlockedHosts allowedAndBlockedHosts) {
+    this.storageService.set(this.initialConnectionHostSpec.getHost(), allowedAndBlockedHosts);
+  }
+
+  @Override
+  public boolean acceptsStrategy(HostRole role, String strategy) throws SQLException {
+    throw new UnsupportedOperationException(
+        Messages.get("PartialPluginService.unexpectedMethodCall", new Object[] {"acceptsStrategy"}));
+  }
+
+  @Override
+  public HostSpec getHostSpecByStrategy(HostRole role, String strategy) throws SQLException {
+    throw new UnsupportedOperationException(
+        Messages.get("PartialPluginService.unexpectedMethodCall", new Object[] {"getHostSpecByStrategy"}));
+  }
+
+  @Override
+  public HostSpec getHostSpecByStrategy(List<HostSpec> hosts, HostRole role, String strategy) throws SQLException {
+    return this.pluginManager.getHostSpecByStrategy(hosts, role, strategy);
+  }
+
+  @Override
+  public HostRole getHostRole(Connection conn) throws SQLException {
+    return this.hostListProvider.getHostRole(conn);
+  }
+
+  private HostSpec getWriter(final @NonNull List<HostSpec> hosts) {
+    for (final HostSpec hostSpec : hosts) {
+      if (hostSpec.getRole() == HostRole.WRITER) {
+        return hostSpec;
+      }
+    }
+    return null;
+  }
+
+  @Override
+  @Deprecated
```

Comment:
What's the point of marking unreleased method as deprecated?

Reply:
This method is deprecated in the regular plugin service (PluginServiceImpl) and shouldn't be used anywhere. Is it better to leave out the deprecated tag since its technically an unreleased method for this particular plugin service, or to keep the deprecated tag since it is deprecated in the regular plugin service and it should not be used? Regardless, I will change the implementation to throw an UnsupportedOperationException like the other methods we don't want called